cmd/mirror-main.go

```diff
@@ -587,6 +591,9 @@ func (mj *mirrorJob) mirror(ctx context.Context, cancelMirror context.CancelFunc
 			}
 		}
 		if err != nil {
+			if err.ToGoError().Error() == ErrBucketDoNotExist.Error() {
```

Comment:
@Praveenrajmani if you see little below, the code already returns `err.Trace()` for errors that we don't ignore. 

What's happening is that the code is always running case `overwriteNotAllowedErr` even for bucket is not found error.

The reason is that `overwriteNotAllowedErr` satisfies error interface that's why that switch case is always executed.

So to avoid this behavior, you can apply the following diff:

```diff
diff --git a/cmd/typed-errors.go b/cmd/typed-errors.go
index fd1590e8..50665be4 100644
--- a/cmd/typed-errors.go
+++ b/cmd/typed-errors.go
@@ -96,11 +96,13 @@ var errInvalidTarget = func(URL string) *probe.Error {
        return probe.NewError(invalidTargetErr(errors.New(msg))).Untrace()
 }

-type overwriteNotAllowedErr error
+type overwriteNotAllowedErr struct {
+       error
+}

 var errOverWriteNotAllowed = func(URL string) *probe.Error {
        msg := "Overwrite not allowed for `" + URL + "`. Use `--overwrite` to override this behavior."
-       return probe.NewError(overwriteNotAllowedErr(errors.New(msg)))
+       return probe.NewError(overwriteNotAllowedErr{errors.New(msg)})
 }

 type sourceIsDirErr error
```

So instead of adding this code, you can modify overwriteNotAllowedErr from interface to structure:

Reply:
Yes `overwriteNotAllowedErr` in switch case acts as default since it is an interface that returns an `error` , So the idea was to string check. 
This seems better though , will change it